pkg/virt-controller/watch/migration.go

```diff
@@ -157,6 +157,49 @@ func ensureSelectorLabelPresent(migration *virtv1.VirtualMachineInstanceMigratio
 	}
 }
 
+func (c *MigrationController) patchVMI(origVMI, newVMI *virtv1.VirtualMachineInstance) error {
+	var ops []string
+
+	if !reflect.DeepEqual(origVMI.Status.MigrationState, newVMI.Status.MigrationState) {
+		newState, err := json.Marshal(newVMI.Status.MigrationState)
+		if err != nil {
+			return err
+		}
+		if origVMI.Status.MigrationState == nil {
+			ops = append(ops, fmt.Sprintf(`{ "op": "add", "path": "/status/migrationState", "value": %s }`, string(newState)))
```

Comment:
Just double checking, it's been a while. Add will fail if an entry exists, so this should be race-condition free, correct?

Reply:
Yes.